### PR TITLE
tpm_device: check_pcrbanks only on new version

### PR DESCRIPTION
**Pull request by yanqzhan**
Only check_pcrbanks when libvirt>=7.10.0 and swtpm>=0.7.0.

Signed-off-by: Yanqiu Zhang <yanqzhan@redhat.com>

### CODE REVIEW

**Comment by yanqzhan** [comment]
Test1: on new version
libvirt-8.0.0-5.module+el8.6.0+14480+c0a3aa0f.x86_64
swtpm-0.7.0-1.20211109gitb79fd91.module+el8.6.0+14480+c0a3aa0f.x86_64

#avocado run --vt-type libvirt pcrbank_default.basic
JOB ID     : a8a5da6a711cf902e5ec8fbb1483cd9f1e478c49
JOB LOG    : /root/avocado/job-results/job-2022-05-04T23.15-a8a5da6/job.log
 (1/2) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.normal_test.default_model.emulator.plain.pcrbank_default.basic: PASS (8.79 s)
 (2/2) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.normal_test.default_model.emulator.encrypted.pcrbank_default.basic: PASS (8.85 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 18.96 s

**Comment by yanqzhan** [comment]
Test2: on old version
libvirt-7.0.0-14.1.module+el8.4.0+11095+d46acebf.x86_64
swtpm-0.4.2-1.20201201git2df14e3.module+el8.4.0+9341+96cf2672.x86_64

#avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 tpm_device..pcrbank_default.basic --vt-connect-uri qemu:///system
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : ae0f13b2021a3912e9eee4afde9afd478047703e
JOB LOG    : /var/lib/avocado/job-results/job-2022-05-05T04.10-ae0f13b/job.log
 (1/2) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.normal_test.tpm-crb_model.emulator.plain.pcrbank_default.basic: PASS (93.12 s)
 (2/2) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.normal_test.tpm-crb_model.emulator.encrypted.pcrbank_default.basic: PASS (112.03 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-05-05T04.10-ae0f13b/results.html
JOB TIME   : 206.71 s

File: libvirt/tests/src/virtual_device/tpm_device.py [hunk 1]
```diff
@@ -127,6 +127,18 @@ def run(test, params, env):
         if not utils_package.package_install(["swtpm", "swtpm-tools"]):
             test.error("Failed to install swtpm swtpm-tools on host")
 
+    def swtpm_version(major, minor):
+        """
+        Get swtpm pkg version and check whether > major.minor
+
+        :return: boolean value compared to major.minor
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_device/tpm_device.py:134]: Please add docstring for params

File: libvirt/tests/src/virtual_device/tpm_device.py [hunk 2]
```diff
@@ -127,6 +127,18 @@ def run(test, params, env):
...
+        """
+        cmd = 'rpm -q swtpm'
+        v_swtpm = process.run(cmd).stdout_text.strip().split('-')
+        v_major = int(v_swtpm[1].split('.')[0])
+        v_minor = int(v_swtpm[1].split('.')[1])
+        return False if (v_major == major and v_minor < minor) else True
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_device/tpm_device.py:140]: Should we think about the version that major > 0 ?

**Review by chunfuwen — APPROVED** [review]
Better change name to compare_swtpm_version
LGTM

File: libvirt/tests/src/virtual_device/tpm_device.py [hunk 3]
```diff
@@ -127,6 +127,18 @@ def run(test, params, env):
         if not utils_package.package_install(["swtpm", "swtpm-tools"]):
             test.error("Failed to install swtpm swtpm-tools on host")
 
+    def swtpm_version(major, minor):
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_device/tpm_device.py:130]: Better change the name to compare_swtpm_version

File: libvirt/tests/src/virtual_device/tpm_device.py (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/src/virtual_device/tpm_device.py:136]: Done. Thanks.

File: libvirt/tests/src/virtual_device/tpm_device.py (hunk 3 above)
Comment (yanqzhan) [reply libvirt/tests/src/virtual_device/tpm_device.py]: Done.

**Review by kylazhang — APPROVED** [review]

**Pull request merged**